### PR TITLE
Cell decorators in GridViewColumn

### PR DESCRIPTION
**Pull request by tomasherceg**
This PR adds support for `CellDecorators` property requested in #1064. 
There will be a separate PR with support for `Validator` properties in concrete columns (`GridViewTextColumn` and others).

### CODE REVIEW

File: src/DotVVM.Framework/Controls/GridView.cs [hunk 1]
```diff
@@ -94,6 +94,21 @@ public GridViewFilterPlacement FilterPlacement
         public static readonly DotvvmProperty RowDecoratorsProperty =
             DotvvmProperty.Register<List<Decorator>?, GridView>(c => c.RowDecorators);
 
+        /// <summary>
+        /// Gets or sets a list of decorators that will be applied on the header row.
+        /// </summary>
+        [MarkupOptions(AllowBinding = false, MappingMode = MappingMode.InnerElement)]
+        [ControlPropertyBindingDataContextChange("DataSource")]
+        [CollectionElementDataContextChange(1)]
```
Comment (exyi) [inline src/DotVVM.Framework/Controls/GridView.cs:102]: Header rows don't have this data context.

File: src/DotVVM.Framework/Controls/GridViewColumn.cs [hunk 2]
```diff
@@ -127,6 +128,47 @@ public bool Visible
         public static readonly DotvvmProperty VisibleProperty
             = DotvvmProperty.Register<bool, GridViewColumn>(c => c.Visible, true);
 
+        /// <summary>
+        /// Gets or sets a list of decorators that will be applied on each cell which is not in the edit mode.
+        /// </summary>
+        [PopDataContextManipulation]
```
Comment (exyi) [inline src/DotVVM.Framework/Controls/GridViewColumn.cs:134]: Cell decorators shouldn't pop, they are in the row context.

**Review by exyi — COMMENTED** [review]
Would you please add test for bindings in the CellDecorators? I'd like to be sure that we are right with the data context types. You can also use this unit testing approach, so we don't rely on UI tests for things that have nothing to do with JS: https://github.com/riganti/dotvvm/blob/main/src/DotVVM.Framework.Tests/ControlTests/SimpleControlTests.cs#L110

File: src/DotVVM.Framework/Controls/GridViewColumn.cs (hunk 2 above)
Comment (tomasherceg) [reply src/DotVVM.Framework/Controls/GridViewColumn.cs]: Good catch, thanks!

File: src/DotVVM.Framework/Controls/GridView.cs (hunk 1 above)
Comment (exyi) [reply src/DotVVM.Framework/Controls/GridView.cs:102]: I think this is still a problem and it's not covered by the test

**Review by exyi — APPROVED** [review]
:+1:

**Comment by GrabzIt** [comment]
Great framework by the way. However is this feature working correctly?

When I use: 

            <dot:GridViewTemplateColumn>
                <CellDecorators>
                    <dot:Decorator class="col-1" />
                </CellDecorators>
                <ContentTemplate>
                    {{value: HTML}}
                </ContentTemplate>
            </dot:GridViewTemplateColumn>

I get:

<td>
    <celldecorators>
        
    </celldecorators>
    <contenttemplate class="col-1">
        Everything is up and running.
    </contenttemplate>
</td>

I am using version 3.2.1. It could be that I am misunderstanding how this should work I was expecting something like.

<td class="col-1">
        Everything is up and running.
</td>

**Comment by quigamdev** [comment]
Hello @GrabzIt, this feature is in DotVVM 4.0. We have released a preview version of it. So you can check out the NuGet.

**Comment by GrabzIt** [comment]
Thanks, that explains it!

**Pull request merged**